message/ipldcbor/schema.ipldsch

```diff
@@ -0,0 +1,78 @@
+# TODO(mvdan): upstream must support Any sooner than later
+type Any union {
+	| Bool   bool
+	| Int    int
+	| Float  float
+	| String string
+	| Bytes  bytes
+	# TODO(mvdan): these are missing from upstream because of Any, too
+	# | Map    map
+	# | List   list
+	# | Link   link
+} representation kinded
+
+type GraphSyncExtensions {String:Any}
+type GraphSyncRequestID int
+type GraphSyncPriority int
+
+type GraphSyncMetadatum struct {
+  link         Link # TODO(mvdan): support "&" links in the ipld-prime schema loader
+  blockPresent Bool
+} representation tuple
+
+type GraphSyncMetadata [GraphSyncMetadatum]
+
+type GraphSyncResponseCode enum {
+  # Informational Codes (request in progress)
+
+  | RequestAcknowledged ("10")
+  | AdditionalPeers ("11")
+  | NotEnoughGas ("12")
+  | OtherProtocol ("13")
+  | PartialResponse ("14")
+  | RequestPaused ("15")
+
+  # Success Response Codes (request terminated)
+
+  | RequestCompletedFull ("20")
+  | RequestCompletedPartial ("21")
+
+  # Error Response Codes (request terminated)
+
+  | RequestRejected ("30")
+  | RequestFailedBusy ("31")
+  | RequestFailedUnknown ("32")
+  | RequestFailedLegal ("33")
+  | RequestFailedContentNotFound ("34")
+  | RequestCancelled ("35")
+} representation int
+
+type GraphSyncRequest struct {
+  id          GraphSyncRequestID  (rename "ID")   # unique id set on the requester side
+  root        Link                (rename "Root") # a CID for the root node in the query
+  # TODO(mvdan): must we copy-paste the selector schema here?
```

Comment:
FYI @hannahhoward, not sure how we want to deal with this.

Reply:
Ideally we would reuse the Go types or loaded schemas from a selector package in ipld-prime, though A) there's no such mechanism to mix schemas right now, B) I don't believe the selector package exposes these, and C) I don't believe the selector package uses bindnode either.

Copy-paste seems like the easiest solution for now. We could swap it out with a reference to a library later on, without changing the wire format.

Reply:
Let's copy paste the selector scheme here for now, just make sure the comments make clear what it is and where it comes from so it's easy to update it.

Can ipld-prime handle codegen from multiple .ipldsch files? If so, that would be a good way to separate it out.

Reply:
I'll definitely have to do some copy-pasting for now. Your idea about multiple files is interesting; I'll see what I can do.

Reply:
This goes back to the schemas-as-docs thing as well; consider https://github.com/ipld/ipld/tree/master/_legacy/specs/data-structures/filecoin where you have a massive amount of schema, you also want to use it for documentation, and putting it in a single file doesn't make sense. go-ipld-schema and js-ipld-schema were made to handle multiple inputs and mush the together so you could have this nice splitting. They also do the .md input, scanning for `ipldsch` blocks to extract, so you can do the docs thing.

Reply:
Resolving; we ended up agreeing on just using `Any` for now. We can make the schema stricter down the line by actually using the selector schema.